argilla/docs/how_to_guides/export.md

```diff
@@ -0,0 +1,156 @@
+---
+description: In this section, we will provide a step-by-step guide to show how to filter and query a dataset.
+---
+
+# Exporting from Argilla
+
+This guide provides an overview of how to export your dataset or its records to Python, your local disk, or the Hugging Face Hub.
+
+In Argilla, you can export two main components of a dataset:
+- The dataset's complete configuration defined in `rg.Settings`. This is useful if your want to share your feedback task or restore it later in Argilla.
+- The records stored in the dataset, including `Metadata`, `Vectors`, `Suggestions`, and `Responses`. This is useful if you want to use your dataset's records outside of Argilla.
+
+# Export an `rg.Dataset` from Argilla
+
+First, we will go through exporting a complete dataset from Argilla. This includes the dataset's configuration and records. All of these methods use the `rg.Dataset.from_*` and `rg.Dataset.to_*` methods.
+
+## Push an Argilla dataset to the Hugging Face Hub
+
+You can push a dataset from Argilla to the Hugging Face Hub. This is useful if you want to share your dataset with the community or version control it. You can push the dataset to the Hugging Face Hub using the `rg.Dataset.to_hub` method.
+
+```python
+import argilla as rg
+
+client = rg.Argilla(api_url="<api_url>", api_key="<api_key>")
+
+workspace = client.workspaces("my_workspace")
+
+dataset = client.datasets(name="my_dataset", workspace=workspace)
+
+dataset.to_hub(repo_id="<repo_id>")
+```
+
+This will push the dataset's `Settings` and records to the hub. If you only want to push the dataset's configuration, you can set the `with_records` parameter to `False`.
+
+```python
+dataset.to_hub(repo_id="<repo_id>", with_records=False)
+```
+
+## Pull a dataset from the Hugging Face Hub
+
+You can pull a dataset from the Hugging Face Hub to Argilla. This is useful if you want to restore a dataset's configuration. You can pull the dataset from the Hugging Face Hub using the `rg.Dataset.from_hub` method.
+
+```python
+
+import argilla as rg
+
+client = rg.Argilla(api_url="<api_url>", api_key="<api_key>")
+
+dataset = rg.Dataset.from_hub(repo_id="<repo_id>")
+```
+
+Note that this approach loads the configuration from the repo and downloads the records. If you only want to load records, use the `load_dataset` method of the `datasets` package, and pass the dataset to `rg.Dataset.log` method. See the [guide on records](record.md) for more information.
+
+## Saving a dataset to a local disk
+
+You can save a dataset from Argilla to your local disk. This is useful if you want to back up your dataset. You can use the `rg.Dataset.to_disk` method.
```

Comment:
Mention that it's highly recommended to do it in an empty directory. Linked to that I also created [this issue](https://github.com/argilla-io/argilla/issues/5185).